### PR TITLE
add jupyterlab notebooks

### PR DESCRIPTION
**Pull request by ebattat**
Improvements:

1. Add jupyterlab notebooks: elasticsearch_operations and analyze_prometheus_logs
2. Add support classes for notebooks: elasticsearch_widgets.py and analyze_prometheus_logs.py
3. Add required packages for setup.py and requirement
4. Add notebooks files to manifest.ini

### CODE REVIEW

File: jupyterlab/templates/elasticsearch_operations/elasticsearch_widgets.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            start_time = start_time_picker.value or time(hour=0, minute=0, second=0)
+            end_time = end_time_picker.value or time(hour=0, minute=0, second=0)
+            start_datetime = datetime.combine(start_date, start_time)
+            end_datetime = datetime.combine(end_date, end_time)
+
+        start_date_picker.observe(on_value_change, names='value')
+        end_date_picker.observe(on_value_change, names='value')
+        start_time_picker.observe(on_value_change, names='value')
+        end_time_picker.observe(on_value_change, names='value')
+        return start_datetime, end_datetime
```
Comment (RobertKrawitz) [inline jupyterlab/templates/elasticsearch_operations/elasticsearch_widgets.py:66]: If I'm interpreting the Github display correctly, this file needs a trailing newline.

**Review by RobertKrawitz — CHANGES_REQUESTED** [review]
One trivial change (need a trailing newline in elasticsearch_widgets.py).

File: jupyterlab/templates/elasticsearch_operations/elasticsearch_widgets.py (hunk 1 above)
Comment (RobertKrawitz) [reply jupyterlab/templates/elasticsearch_operations/elasticsearch_widgets.py:66]: I would like @athiruma to review this also, as I'm not familiar with this area.

Comment (ebattat) [reply jupyterlab/templates/elasticsearch_operations/elasticsearch_widgets.py:66]: @RobertKrawitz, its a jupyter widgets method, I test on jupyterlab and it works fine

Comment (RobertKrawitz) [reply jupyterlab/templates/elasticsearch_operations/elasticsearch_widgets.py:66]: There's no issue with the code, but Linux (POSIX in general) text files are required to always have a trailing newline, reference https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_206  ("A sequence of zero or more non- <newline> characters plus a terminating <newline> character.") and https://gist.github.com/OleksiyRudenko/d51388345ea55767b7672307fe35adf3

File: jupyterlab/templates/logs_analyzer/analyze_prometheus_logs.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        # download with progress bar
+        response = session.get(self.__s3_logs_url, stream=True)
+        size = int(response.headers.get('Content-Length', 0))
+
+        progress = widgets.IntProgress(description='Downloading', min=0, max=size)
+        display(progress)
+
+        with open(f"{self.__logs_dir}/{self.__filename}", 'wb') as f:
+            with tqdm.wrapattr(f, "write", total=size) as fileobj:
+                for chunk in response.iter_content(chunk_size=8192):
```
Comment (athiruma) [inline jupyterlab/templates/logs_analyzer/analyze_prometheus_logs.py:69]: I think chunk size should be const

File: jupyterlab/templates/elasticsearch_operations/elasticsearch_widgets.py (hunk 1 above)
Comment (ebattat) [reply jupyterlab/templates/elasticsearch_operations/elasticsearch_widgets.py]: @RobertKrawitz, right update it.

**Review by RobertKrawitz — APPROVED** [review]

**Comment by RobertKrawitz** [comment]
I've approved it.  I'm not marking it approved pending @athiruma approval.

File: jupyterlab/templates/logs_analyzer/analyze_prometheus_logs.py (hunk 2 above)
Comment (ebattat) [reply jupyterlab/templates/logs_analyzer/analyze_prometheus_logs.py:69]: @athiruma, good catch, fixed it

File: jupyterlab/templates/logs_analyzer/analyze_prometheus_logs.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        logger.info(f'untar download file {self.__log_dir_path}')
+        os.system(f"tar -xvf {self.__log_dir_path} -C {self.__logs_dir}")
+
+        promdb_file = [f for f in os.listdir(self.__log_dir_path.split('.')[0]) if f.startswith('promdb')]
+
+        logger.info(f"untar prometheus file: {os.path.join(self.__logs_dir, self.__filename.split('.')[0], promdb_file[0])}")
+        os.system(f"tar -xvf {os.path.join(self.__logs_dir, self.__filename.split('.')[0], promdb_file[0])} -C {os.path.join(self.__logs_dir, self.__filename.split('.')[0])}")
+
+        promdb_file = [f for f in os.listdir(self.__log_dir_path.split('.')[0]) if f.startswith('promdb') and not f.endswith('tar')]
+        promdb_dir_path = f"{os.path.join(self.__logs_dir, self.__filename.split('.')[0], promdb_file[0])}"
```
Comment (athiruma) [inline jupyterlab/templates/logs_analyzer/analyze_prometheus_logs.py:91]: Zeros are repeated. Can we make it as const?

File: jupyterlab/templates/logs_analyzer/analyze_prometheus_logs.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+    def open_grafana_dashboard(self, promdb_dir_path: str, grafana_dashboard_url: str):
+        """
+        This method opens the Grafana dashboard that is mounted to promdb
+        @param promdb_dir_path:
+        @param grafana_dashboard_url:
+        @return:
+        """
+        start_time = promdb_dir_path.split('/')[-1].split('+')[:2][0].replace('prom', '').replace('db_', '')
+        end_time = promdb_dir_path.split('/')[-1].split('+')[:2][1].replace('prom', '').replace('0000_', '')
```
Comment (athiruma) [inline jupyterlab/templates/logs_analyzer/analyze_prometheus_logs.py:131]: I think, It is better to specify the input parameters format.

File: jupyterlab/templates/logs_analyzer/analyze_prometheus_logs.py (hunk 3 above)
Comment (ebattat) [reply jupyterlab/templates/logs_analyzer/analyze_prometheus_logs.py:91]: Yes

**Review by athiruma — APPROVED** [review]
/lgtm

**Pull request merged**